sdk/core/azure-core-test/inc/azure/core/test/test_context_manager.hpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * This class handles managing context about a test.
+ */
+
+#pragma once
+
+#include <gtest/gtest.h>
+#include <string>
+
+#include "azure/core/test/network_models.hpp"
+
+namespace Azure { namespace Core { namespace Test {
+  class TestContextManager {
+  public:
+    std::string RecordingPath;
+    Azure::Core::Test::TestMode TestMode;
+    bool LiveOnly = false;
+
+    TestContextManager() { SetLiveOnly(); }
+
+    void RenameTest(std::string const& testName)
+    {
+      m_testName = testName;
+      SetLiveOnly();
+    }
+
+    void RenameTest(std::string const& testSuite, std::string const& testName)
+    {
+      m_testSuite = testSuite;
+      RenameTest(testName);
+    }
+
+    /**
+     * @brief The test suite name plus the test name.
+     *
+     * @return m_testSuiteName.m_testName
+     */
+    std::string GetTestPlaybackRecordingName() const
+    {
+      std::string fullName(m_testSuite);
+      fullName.append(".");
+      fullName.append(m_testName);
+
+      return fullName;
+    }
+
+    /**
+     * @brief Get the Test Mode object for the current test.
+     *
+     * @return Either LIVE, PLAYBACK or RECORD.
+     */
+    Azure::Core::Test::TestMode GetTestMode() const { return TestMode; }
+
+    /**
+     * @brief Check if test is expected to be recorded.
+     *
+     * @return `true` when test is expected to be recorded or `false` when it is not.
+     */
+    bool DoNotRecordTest() const
+    {
+      return TestMode != Azure::Core::Test::TestMode::RECORD || LiveOnly;
+    }
+
+    bool IsPlaybackMode() const { return TestMode == Azure::Core::Test::TestMode::PLAYBACK; }
+    bool IsLiveMode() const { return TestMode == Azure::Core::Test::TestMode::LIVE; }
+
```

Comment:
we could also add a isRecordMode() for completion

Reply:
Can be added later, sure.